tests/test_browser.py

```diff
@@ -3556,6 +3556,41 @@ def test_dynamic_link_many(self):
     self.btest_exit(self.in_dir('main.c'), '3',
                     args=['-s', 'MAIN_MODULE', '--pre-js', 'pre.js'])
 
+  def test_dynamic_link_pthread_many(self):
+    # test asynchronously loading two side modules during startup
+    # they should always load in the same order
+    create_test_file('main.cpp', r'''
+      #include <assert.h>
+      #include <thread>
+      extern "C" int side1();
+      extern "C" int side2();
+      int main() {
+        auto side1_ptr = &side1;
+        auto side2_ptr = &side2;
+        std::thread([=]{
+          REPORT_RESULT(int(
+            side1_ptr() == 1 &&
+            side2_ptr() == 2
+          ));
+        }).detach();
+        return 0;
+      }
+    ''')
+    # Use a big payload in side1 so that it takes longer to load than side2
+    create_test_file('side1.cpp', r'''
+      char const * payload1 = "''' + str(list(range(1, int(1e5)))) + r'''";
+      extern "C" int side1() { return 1; }
+    ''')
+    create_test_file('side2.cpp', r'''
+      char const * payload2 = "0";
+      extern "C" int side2() { return 2; }
+    ''')
+    self.run_process([EMCC, 'side1.cpp', '-Wno-experimental', '-pthread', '-s', 'SIDE_MODULE', '-o', 'side1.wasm'])
+    self.run_process([EMCC, 'side2.cpp', '-Wno-experimental', '-pthread', '-s', 'SIDE_MODULE', '-o', 'side2.wasm'])
+    self.btest(self.in_dir('main.cpp'), '1',
```

Comment:
It might be good to write new tests using `btest_exit` - is that right @sbc100 ?

Reply:
To use `btest_exit` the test would need to join the thread.
At the same time, it needs to run in the browser's main thread to hit the problem, where it can't join other threads.